app/views/ChooseProvider.js

```diff
@@ -215,33 +199,59 @@ class ChooseProviderScreen extends Component {
     );
   }
 
+  toggleFilterAuthoritesByGPSHistory() {
+    this.filterAuthoritesByGPSHistory({
+      val: !this.state.isAuthorityFilterActive,
+    });
+  }
+
+  async filterAuthoritesByGPSHistory(isAuthorityFilterActive) {
+    await this.fetchAuthoritiesList(isAuthorityFilterActive.val);
+    this.setState({ isAuthorityFilterActive: isAuthorityFilterActive.val });
+  }
+
+  async toggleAutoSubscribe() {
+    this.setState(
+      prevState => ({
+        isAutoSubscribed: !prevState.isAutoSubscribed,
+      }),
+      async () => {
+        this.state.isAutoSubscribed
```

Comment:
not sure on this setState form, but is this.state guaranteed to be up to date? or does this function have a parameter with the latest state since update?

Reply:
Yeah so this is a little hacky but `setState` has an optional callback param that you can pass which will be executed once state is updated. 

https://stackoverflow.com/questions/43370176/using-async-setstate

I think the alternative would be to wrap the state update in a promise and then await that, e.g.

```js
asyncToggleSubscribe() {
  return new Promise((resolve) => this.setState(prevState => ({
        isAutoSubscribed: !prevState.isAutoSubscribed,
      })
}
```

Reply:
All good no change needed. Although if you like making it Async await would be clearer